src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs

```diff
@@ -350,7 +350,10 @@ public object CreateVector(string fieldName, object value)
             // - value is textual or array of textual we're treating them as text input
             // - otherwise, we will write as array of numerical values
             var currentIndexingScope = CurrentIndexingScope.Current;
-
+            if (currentIndexingScope != null && currentIndexingScope.Index.IsOnBeforeExecuteIndexing)
```

Comment:
What is that about?

Reply:
Dictionary training, there is no reason to generate embeddings / create vectors values in training phase.

Reply:
Can we give it a better name, I couldn't understand the intent there.

Reply:
`DictionaryTraining` is hidden in the Corax implementation of `IndexPersistence.OnBeforeExecuteIndexing([...]);`. To maintain consistency, I would leave it as it is; however, I will add an explanatory comment.